homeassistant/components/openrgb/light.py

```diff
@@ -0,0 +1,342 @@
+        # because there is no guarantee that the device won't go back to black
+        need_to_apply_color = (
+            ATTR_RGB_COLOR in kwargs
+            or ATTR_BRIGHTNESS in kwargs
+            or self._attr_brightness is None
+            or self._attr_rgb_color is None
+        )
+
+        # If color/brightness parameters require color support but mode doesn't support it,
+        # switch to a color-capable mode
+        if need_to_apply_color and not mode_supports_color:
+            mode = self._preferred_no_effect_mode
+
+        if mode is not None:
+            await self._async_apply_mode(mode)
+
+        if need_to_apply_color:
+            brightness = None
+            if ATTR_BRIGHTNESS in kwargs:
+                brightness = kwargs[ATTR_BRIGHTNESS]
+            elif self._attr_brightness is None:
+                # Restore previous brightness when turning on
+                brightness = self._previous_brightness
+            if brightness is None:
+                # Use default brightness if still None
+                brightness = DEFAULT_BRIGHTNESS
+
+            color = None
+            if ATTR_RGB_COLOR in kwargs:
+                color = kwargs[ATTR_RGB_COLOR]
+            elif self._attr_rgb_color is None:
+                # Restore previous color when turning on
+                color = self._previous_rgb_color
+            if color is None:
+                # Use default color if still None
+                color = DEFAULT_COLOR
+
+            await self._async_apply_color(color, brightness)
+
+        await self.coordinator.async_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the light."""
+        if self._supports_off_mode:
+            await self._async_apply_mode(OPENRGB_MODE_OFF)
+        else:
+            # If the device does not support Off mode, set color to black
+            await self._async_apply_color(OFF_COLOR, 0)
+
+        await self.coordinator.async_refresh()
+
+
+def check_if_mode_supports_color(mode: ModeData) -> bool:
+    """Return True if the mode supports colors."""
+    if mode.flags & ModeFlags.HAS_PER_LED_COLOR:
+        return True
+    if mode.flags & ModeFlags.HAS_MODE_SPECIFIC_COLOR:
+        return True
+    return False
+
+
+def get_icon(device_type: DeviceType) -> str | None:
+    """Return an icon for the device_type."""
+    icons = {
+        DeviceType.MOTHERBOARD: "developer-board",
+        DeviceType.DRAM: "memory",
+        DeviceType.GPU: "expansion-card",
+        DeviceType.COOLER: "fan",
+        DeviceType.LEDSTRIP: "led-variant-on",
+        DeviceType.KEYBOARD: "keyboard",
+        DeviceType.MOUSE: "mouse",
+        DeviceType.MOUSEMAT: "rug",
+        DeviceType.HEADSET: "headset",
+        DeviceType.HEADSET_STAND: "headset-dock",
+        DeviceType.GAMEPAD: "gamepad-variant",
+        DeviceType.SPEAKER: "speaker",
+        DeviceType.STORAGE: "harddisk",
+        DeviceType.CASE: "desktop-tower",
```

Comment:
make this a `property` of `OpenRGBLight`

```suggestion
    @property
    def icon(self) -> str | None:
        """Return an icon for the device_type."""
        icons = {
            DeviceType.MOTHERBOARD: "developer-board",
            DeviceType.DRAM: "memory",
            DeviceType.GPU: "expansion-card",
            DeviceType.COOLER: "fan",
            DeviceType.LEDSTRIP: "led-variant-on",
            DeviceType.KEYBOARD: "keyboard",
            DeviceType.MOUSE: "mouse",
            DeviceType.MOUSEMAT: "rug",
            DeviceType.HEADSET: "headset",
            DeviceType.HEADSET_STAND: "headset-dock",
            DeviceType.GAMEPAD: "gamepad-variant",
            DeviceType.SPEAKER: "speaker",
            DeviceType.STORAGE: "harddisk",
            DeviceType.CASE: "desktop-tower",
            DeviceType.MICROPHONE: "microphone",
            DeviceType.KEYPAD: "dialpad",
        }

        return icons.get(self.device.type)
```

Reply:
Looks good, except for one small nitpick. _attr_icon is already None by default and if it is None will fall back to the default icon, so you can write just
```
        self._attr_icon = DEVICE_TYPE_ICONS.get(self.device.type)
```

Reply:
Makes sense. Thank you.